recreated deserializeDocument (fixes random bugs)

matanshukry opened the pull request:
The old method was applying the field types as part of the keys.
The old way was problematic, mainly because the order of the keys isn't correct. If the ``@fieldTypes`` key would hit the loop first, it would use empty values and not set anything. If it would come last, it would come after the values has already been parsed (in which case Long.parseString throws errors, since the function assumes the input has .charAt() method, which the parsed number passed to it, doesn't).

lvca commented:
Makes sense. Merging now. Thanks again Matan. @maggiolo00 Could you please double check it?

maggiolo00 commented:
@lvca  i will check it today

maggiolo00 commented:
@matanshukry 

from a first look

```
var fieldTypes = prepareFieldTypes(serialized['@fieldTypes']) || {};
```
it will return always an empty Object since serialized is a String not an Object

matanshukry commented:
You are correct. I found the problem to be bigger than just that, though, and including some more.
I had to create a second pull request, since I continued working on the same branch. I did merge the current master, so you can just merge the new pull request ( #131 ).

Some details:

1. I had to "recreate" JSON.parse. I didn't actually recreate it, but rather took it from the original author and changed it a bit ( https://github.com/douglascrockford/JSON-js )
The problem with the original code is you can only transform values, and not actually parse them. As such, if a number if too big for javascript Number, you can't use orientjs Long class to parse it, since you are already receiving it as a truncated number.

2. I added a test case (I put it under bugs, wasn't sure where to put it), to test the de-serialization process and some special values.

p.s.
Currently the travis ci build fails due to problem with the double number. The number has 10 significant digits, and since double, which is a number in javascript and java.lang.Double in java (orientdb), can handle up to 15 digits of precision without loss, I am not sure where the problem is.

Obviously, the test does work at my pc (Using node v4.2.3 and OrientDB v2.1-rc6)
